ext/ddtrace_profiling_native_extension/http_transport.c

```diff
@@ -0,0 +1,285 @@
+#include <ruby.h>
+#include <ruby/thread.h>
+#include <ddprof/ffi.h>
+
+// Used to report profiling data to Datadog.
+// This file implements the native bits of the Datadog::Profiling::HttpTransport class
+
+// Datadog::Profiling::HttpTransport::Exporter
+// This is used to wrap a native pointer to a libddprof exporter as a Ruby object of this class;
+// see exporter_as_ruby_object below for more details
+static VALUE exporter_class = Qnil;
+
+static VALUE ok_symbol = Qnil; // :ok in Ruby
+static VALUE error_symbol = Qnil; // :error in Ruby
+
+#define byte_slice_from_literal(string) ((ddprof_ffi_ByteSlice) {.ptr = (uint8_t *) "" string, .len = sizeof("" string) - 1})
+
+struct call_exporter_without_gvl_arguments {
+  ddprof_ffi_ProfileExporterV3 *exporter;
+  ddprof_ffi_Request *request;
+  ddprof_ffi_SendResult result;
+};
+
+inline static ddprof_ffi_ByteSlice byte_slice_from_ruby_string(VALUE string);
```

Comment:
FYI, you can probably drop `inline` here; in this case it wouldn't do anything different than just `static`. The only exception I'm aware is when you mix `inline static` and plain `inline` (no static) functions, then sometimes the compiler will yell at you. It's not important, just a bit of a C semantic you may or may not know about :)

Reply:
TIL! I expect at some point this will move to a shared file, and I'll drop the `static` part, so I'm thinking of leaving it for now more as a declaration of intention than expecting different semantics.

Nice link discussing this as well -> https://stackoverflow.com/questions/7762731/whats-the-difference-between-static-and-static-inline-function .